### PR TITLE
Port `CryptoTransferHandler.preHandle()` and sig req tests

### PR DESCRIPTION
**Pull request by tinker-michaelj**
😵‍💫 _Note to reviewer:_ spotless in its wisdom decided to update license dates on a few hundred files here, sorry. The main changes to review are `CryptoTransferHandler.preHandle()` and especially `CryptoTransferHandlerParityTest` with its connected work in `testFixtures`.

**Description**:
 - Moves @georg-getz's work on `CryptoTransfer` pre-handle into its new home (`CryptoTransferHandler.preHandle()`).
 - Ports fixes and `SigRequirements` unit tests into a form compatible with current module structure.
 - Includes unit test memory leak fixes from `release/0.34`; and increases `test` task `maxHeapSize=2048m` in _buildSrc/src/main/kotlin/com.hedera.hashgraph.conventions.gradle.kts_.
 - Closes https://github.com/hashgraph/hedera-services/issues/4420 (all major difficulties in re-using `SigRequirementsTest` scenarios are resolved).

### CODE REVIEW

File: hedera-node/hedera-mono-service/src/testFixtures/java/com/hedera/test/utils/SimpleKeyLookup.java [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+import com.hedera.node.app.service.mono.legacy.core.jproto.JKey;
+import com.hedera.node.app.service.mono.state.migration.HederaAccount;
+import com.hedera.node.app.spi.AccountKeyLookup;
+import com.hedera.node.app.spi.KeyOrLookupFailureReason;
+import com.hedera.node.app.spi.state.State;
+import com.hedera.node.app.spi.state.States;
+import com.hederahashgraph.api.proto.java.AccountID;
+import edu.umd.cs.findbugs.annotations.NonNull;
+
+public class SimpleKeyLookup implements AccountKeyLookup {
```
Comment (mhess-swl) [inline hedera-node/hedera-mono-service/src/testFixtures/java/com/hedera/test/utils/SimpleKeyLookup.java:38]: Why do we need to differentiate between a simple key lookup and a...complicated? key lookup?'

File: gradle.properties [hunk 2]
```diff
@@ -11,4 +11,4 @@ org.gradle.caching = true
 includesRegex = .*
 
 # Require spotless to fetch license header dates from Git history
-spotlessSetLicenseHeaderYearsFromGitHistory = false
\ No newline at end of file
+spotlessSetLicenseHeaderYearsFromGitHistory = false
```
Comment (kimbor) [inline gradle.properties:14]: In the history of this PR, services version # went from 33 to 34 then back to 33. Was the going back to 33 a mistake? It looks to me like it got accidentally merged in when a different change was reverted.
`version = 0.33.0-SNAPSHOT`

**Review by Neeharika-Sompalli — APPROVED** [review]
LGTM ! Need to merge develop back into this branch.

File: hedera-node/hedera-mono-service/src/test/java/com/hedera/node/app/service/mono/state/exports/SignedStateBalancesExporterTest.java [hunk 3]
```diff
@@ -141,6 +140,7 @@ class SignedStateBalancesExporterTest {
 
     @BeforeEach
     void setUp() throws ConstructableRegistryException, NoSuchAlgorithmException {
+        tokenRels = TokenRelStorageAdapter.fromInMemory(new MerkleMap<>());
```
[the remark this answers is not in the document]
Comment (tinker-michaelj) [reply hedera-node/hedera-mono-service/src/test/java/com/hedera/node/app/service/mono/state/exports/SignedStateBalancesExporterTest.java:143]: We could revert this change, it was just a random experiment before we knew the cause of the unit test failures. But it shouldn't matter at all either way.

File: hedera-node/hedera-mono-service/src/testFixtures/java/com/hedera/test/utils/SimpleKeyLookup.java (hunk 1 above)
Comment (tinker-michaelj) [reply hedera-node/hedera-mono-service/src/testFixtures/java/com/hedera/test/utils/SimpleKeyLookup.java:38]: Haha I'll rename to "`TestFixturesKeyLookup`" 😁

File: hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/state/migration/UniqueTokenMapAdapter.java [hunk 4]
```diff
@@ -189,4 +189,9 @@ public void archive() {
     public Hash getHash() {
         return isVirtual ? virtualMap.getHash() : merkleMap.getHash();
     }
+
+    @Nullable
+    public VirtualMap<UniqueTokenKey, UniqueTokenValue> getOnDiskNfts() {
+        return virtualMap;
+    }
```
[the remark this answers is not in the document]
Comment (tinker-michaelj) [reply hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/state/migration/UniqueTokenMapAdapter.java:196]: Right, `ResponsibleVMapUser` needed to be able to access this `VirtualMap`.

File: gradle.properties (hunk 2 above)
Comment (tinker-michaelj) [reply gradle.properties:14]: Hmm yes seems like an accident. 🤔

File: hedera-node/hedera-mono-service/src/test/java/com/hedera/node/app/service/mono/state/exports/SignedStateBalancesExporterTest.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (mhess-swl) [reply hedera-node/hedera-mono-service/src/test/java/com/hedera/node/app/service/mono/state/exports/SignedStateBalancesExporterTest.java:143]: Functionally it won't matter, I agree. I'm thinking of my future self when I come back to this test someday, having forgotten all the now-current but then-past context, and imagining a very confused self asking why it was done for `tokenRels` and not for the others 😂

File: hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/state/migration/UniqueTokenMapAdapter.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (mhess-swl) [reply hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/state/migration/UniqueTokenMapAdapter.java:196]: > Right, `ResponsibleVMapUser` needed to be able to access this `VirtualMap`.
Let me clarify what I meant, not very obvious–the class you added the virtual map getter to, UniqueTokenMapAdapter, _already has an accessor for the underlying virtual map_–it’s just not called `nftsOnDisk` (or similar), it's called `adapter.virtualMap()`. Did you add another accessor for the convenience of being able to get that virtual map by the same designation as the other two adapters (i.e. xxxOnDisk), or was there a different reason to duplicate the original virtual map's getter?

File: hedera-node/hedera-mono-service/src/testFixtures/java/com/hedera/test/utils/SimpleKeyLookup.java (hunk 1 above)
Comment (mhess-swl) [reply hedera-node/hedera-mono-service/src/testFixtures/java/com/hedera/test/utils/SimpleKeyLookup.java:38]: Ahhh, now the context makes much more sense, thank you!

**Review by kimbor — APPROVED** [review]
looks good!

**Pull request merged**